### PR TITLE
feat: add parallel modes file to pipeline

### PR DESCRIPTION
**Pull request by cyw233**
### Description of PR

Add parallel modes file option to pipeline so users can customize the parallel modes by using their own file in the `test_parallel_modes/` folder. 

Summary:
Fixes # (issue) Microsoft ADO 29867650

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
We want users to be able to customize the parallel modes file, so we added the parallel modes file option to the pipeline.

### CODE REVIEW

File: .azure-pipelines/run-test-elastictest-template.yml [hunk 1]
```diff
@@ -119,6 +119,10 @@ parameters:
     type: string
     default: ""
 
+  - name: PARALLEL_MODES_FILE
+    type: string
+    default: "default.json"
```
Comment (lerry-lee) [inline .azure-pipelines/run-test-elastictest-template.yml:124]: It is recommended to leave this parameter empty, as parallel run is disabled by default and we already have a default value in test_plan.py and  elastictest. Additionally, consider adding a comment for clarification if necessary.

Comment (cyw233) [reply .azure-pipelines/run-test-elastictest-template.yml:124]: Good idea! Let me fix this

**Review by lerry-lee — APPROVED** [review]
LGTM

**Comment by cyw233** [comment]
> @cyw233 PR conflicts with 202405 branch

No need to cherry-pick the entire PR to feature branch. We only need to cherry-pick the parallel mode files to 202405: #16134

**Comment by cyw233** [comment]
Cherry-pick the parallel mode files to 202411: #16323

**Pull request merged**